pkg/receive/handler.go

```diff
@@ -282,13 +283,17 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	span, ctx := tracing.StartSpan(r.Context(), "receive_http")
 	defer span.Finish()
 
-	compressed, err := ioutil.ReadAll(r.Body)
+	compressed := &bytes.Buffer{}
+	if r.ContentLength >= 0 {
+		compressed.Grow(int(r.ContentLength))
+	}
+	_, err := io.Copy(compressed, r.Body)
```

Comment:
Is this necessary? I am curious how much does it optimize?

Reply:
I will comment (: